python/feathub/feature_views/derived_feature_view.py

```diff
@@ -171,28 +190,44 @@ def _get_feature_from_feature_str(
             return feature
         elif len(parts) == 2:
             join_table_name = parts[0]
-            join_feature_name = parts[1]
             table_desc = registry.get_features(
                 name=join_table_name, force_update=force_update
             )
-            join_feature = table_desc.get_feature(feature_name=join_feature_name)
-            if join_feature.keys is None:
-                raise RuntimeError(
-                    f"Feature '{join_feature_name}' in the remote table "
-                    f"'{join_table_name}' does not have keys specified."
+            if is_id(parts[1]):
+                join_feature_name = parts[1]
+                join_feature = table_desc.get_feature(feature_name=join_feature_name)
+                if join_feature.keys is None:
+                    raise RuntimeError(
+                        f"Feature '{join_feature_name}' in the remote table "
+                        f"'{join_table_name}' does not have keys specified."
+                    )
+                return Feature(
+                    name=join_feature_name,
+                    dtype=join_feature.dtype,
+                    transform=JoinTransform(join_table_name, join_feature_name),
+                    keys=join_feature.keys,
                 )
-            return Feature(
-                name=join_feature_name,
-                dtype=join_feature.dtype,
-                transform=JoinTransform(join_table_name, join_feature_name),
-                keys=join_feature.keys,
-            )
-        else:
-            raise FeathubException(
-                "Invalid string format. If a feature is a string, it should be either "
-                "in the format {table_name}.{feature_name} or in the "
-                "format {feature_name}."
-            )
+            elif is_static_map_lookup_op(parts[1]):
+                join_feature_expr = parts[1]
+                variable_types: Dict[str, Optional[DType]] = {
+                    f.name: f.dtype for f in table_desc.get_output_features()
+                }
+                return Feature(
+                    name=default_feature_name,
+                    dtype=cast(
```

Comment:
Since the logic of determing the dtype is non-trivial (involving 3+ function calls and spread across 3 lines), would it be more readable to determine it before constructing the Feature instance?